### PR TITLE
common: avoid useless uses of

### PR DESCRIPTION
**Pull request by maxice8**

### CODE REVIEW

**Comment by xtraeme** [comment]
We should use $XBPS_DIGEST_CMD too, rather than sha256sum(1)

**Comment by xtraeme** [comment]
This is mainly to avoid external deps as few as possible. xbps-uhelper is part of xbps, while sha256sum is part of coreutils.

**Comment by maxice8** [comment]
On Sat, Apr 13, 2019 at 11:03:24AM -0700, Juan RP wrote:
> This is mainly to avoid external deps as few as possible. xbps-uhelper is part of xbps, while sha256sum is part of coreutils. 
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/10906#issuecomment-482853944

Thanks for the tip, i am changing it

**Comment by maxice8** [comment]
I like how it avoids extra calls to `awk` and `cut`

**Comment by xtraeme** [comment]
I'd like to share my latest changes to xbps-src, hopefully in the upcoming week.

**Comment by xtraeme** [comment]
We still need internally some cmds provided by third party, like sed, flock and others.

File: common/hooks/do-patch/00-patches.sh [hunk 1]
```diff
@@ -39,13 +39,13 @@ hook() {
 		return 0
 	fi
 	if [ -r $PATCHESDIR/series ]; then
-		cat $PATCHESDIR/series | while read f; do
+		while read -r f; do
```
Comment (maxice8) [inline common/hooks/do-patch/00-patches.sh]: We should problably document the existance of this.

**Comment by xtraeme** [comment]
Yeah, it has been there for years. It's to specify the order to apply the patches.

**Comment by maxice8** [comment]
On Sun, Apr 14, 2019 at 12:22:59AM -0700, Juan RP wrote:
> Yeah, it has been there for years. It's to specify the order to apply the patches. 
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/10906#issuecomment-482926511

I'll write a ### Handling patches section for the Manual.md, seems like xbps-src has lots of undocumented areas and how it works internally is also hard to follow.

File: common/hooks/do-patch/00-patches.sh (hunk 1 above)
Comment (maxice8) [reply common/hooks/do-patch/00-patches.sh]: #10922

**Comment by xtraeme** [comment]
Done

File: common/hooks/pre-configure/02-script-wrapper.sh [hunk 2]
```diff
@@ -8,8 +8,11 @@ generic_wrapper() {
 	[ ! -x ${XBPS_CROSS_BASE}/usr/bin/${wrapper} ] && return 0
 	[ -x ${XBPS_WRAPPERDIR}/${wrapper} ] && return 0
 
-	echo "#!/bin/sh" >> ${XBPS_WRAPPERDIR}/${wrapper}
-	echo "exec ${XBPS_CROSS_BASE}/usr/bin/${wrapper} --prefix=${XBPS_CROSS_BASE}/usr \"\$@\"" >> ${XBPS_WRAPPERDIR}/${wrapper}
+	cat >>${XBPS_WRAPPERDIR}/${wrapper}<<_EOF
```
Comment (congdanhqx) [inline common/hooks/pre-configure/02-script-wrapper.sh]: I think it's cleaner to append minus `-` before `_EOF`,
we can indent the here doc with tab then.

Comment (maxice8) [reply common/hooks/pre-configure/02-script-wrapper.sh]: No, that is inconsistent with all other usages in the file.

**Comment by xtraeme** [comment]
Why didn't you merge this?

**Comment by maxice8** [comment]
On Mon, Apr 15, 2019 at 01:35:13AM -0700, Juan RP wrote:
> Why didn't you merge this? 
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/10906#issuecomment-483159744

Doing other stuff at the moment

**Comment by xtraeme** [comment]
Please merge this :-)

**Comment by maxice8** [comment]
On Mon, Apr 15, 2019 at 07:43:43AM -0700, Juan RP wrote:
> Please merge this :-)
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/10906#issuecomment-483281381

Finish my cup noddles for me then
